components/widgets/menu/menu_items/menu_item_toggle_modal_redux.jsx

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ToggleModalButtonRedux from 'components/toggle_modal_button_redux';
+
+import menuItem from './menu_item.jsx';
+
+export const MenuItemToggleModalReduxImpl = ({modalId, dialogType, dialogProps, text}) => (
```

Comment:
Looks like this is (indirectly) connected to the redux store, which means strictly speaking it shouldn't be in the `widgets` library. Having said that, I see no harm in it staying here for now, and we can move it elsewhere when we get properly stuck in to the "Widget Library" work in a few months.

Reply:
Was for simplicity, but I would possible to remove this kind of items and use MenuItemAction whith the behavior defined in the parent component of MenuWrapper for each case.

Reply:
OK makes sense. I'm happy for it to stay part of this PR for simplicity.